pkg/collector/corechecks/cluster/kubernetes_state_aggregators.go

```diff
@@ -0,0 +1,152 @@
+		log.Error("BUG in KSM metric aggregator")
+		return nil
+	}
+
+	return &sumValuesAggregator{
+		counterAggregator{
+			metricName:    metricName,
+			allowedLabels: allowedLabels,
+			accumulator:   make(map[[maxNumberOfAllowedLabels]string]float64),
+		},
+	}
+}
+
+func newCountObjectsAggregator(metricName string, allowedLabels []string) metricAggregator {
+	if len(allowedLabels) > maxNumberOfAllowedLabels {
+		// `maxNumberOfAllowedLabels` is hardcoded to the maximum number of labels passed to this function from the metricsAggregators definition below.
+		// The only possibility to arrive here is to add a new aggregator in `metricAggregator` below and to forget to update `maxNumberOfAllowedLabels` accordingly.
+		log.Error("BUG in KSM metric aggregator")
+		return nil
+	}
+
+	return &countObjectsAggregator{
+		counterAggregator{
+			metricName:    metricName,
+			allowedLabels: allowedLabels,
+			accumulator:   make(map[[maxNumberOfAllowedLabels]string]float64),
+		},
+	}
+}
+
+func (a *sumValuesAggregator) accumulate(metric ksmstore.DDMetric) {
+	var labelValues [maxNumberOfAllowedLabels]string
+
+	for i, allowedLabel := range a.allowedLabels {
+		if allowedLabel == "" {
+			break
+		}
+
+		labelValues[i] = metric.Labels[allowedLabel]
+	}
+
+	a.accumulator[labelValues] += metric.Val
+}
+
+func (a *countObjectsAggregator) accumulate(metric ksmstore.DDMetric) {
+	var labelValues [maxNumberOfAllowedLabels]string
+
+	for i, allowedLabel := range a.allowedLabels {
+		if allowedLabel == "" {
+			break
+		}
+
+		labelValues[i] = metric.Labels[allowedLabel]
+	}
+
+	a.accumulator[labelValues]++
+}
+
+func (a *counterAggregator) flush(sender aggregator.Sender, k *KSMCheck, labelJoiner *labelJoiner) {
+	for labelValues, count := range a.accumulator {
+
+		labels := make(map[string]string)
+		for i, allowedLabel := range a.allowedLabels {
+			if allowedLabel == "" {
+				break
+			}
+
+			labels[allowedLabel] = labelValues[i]
+		}
+
+		hostname, tags := k.hostnameAndTags(labels, labelJoiner)
+
+		sender.Gauge(ksmMetricPrefix+a.metricName, count, hostname, tags)
+	}
+
+	a.accumulator = make(map[[maxNumberOfAllowedLabels]string]float64)
+}
+
+var (
+	metricAggregators = map[string]metricAggregator{
+		"kube_persistentvolume_status_phase": newSumValuesAggregator(
+			"persistentvolumes.by_phase",
+			[]string{"storageclass", "phase"},
+		),
+		"kube_service_spec_type": newCountObjectsAggregator(
+			"service.count",
+			[]string{"namespace", "type"},
+		),
+		"kube_namespace_status_phase": newSumValuesAggregator(
+			"namespace.count",
+			[]string{"phase"},
+		),
+		"kube_replicaset_owner": newCountObjectsAggregator(
+			"replicaset.count",
+			[]string{"namespace", "owner_name", "owner_kind"},
+		),
+		"kube_job_owner": newCountObjectsAggregator(
+			"job.count",
+			[]string{"namespace", "owner_name", "owner_kind"},
+		),
+		"kube_deployment_status_observed_generation": newCountObjectsAggregator(
```

Comment:
I just found out that we exclude `*_generation` metrics in the exclude list defined in `kubernetes_state_defaults.go`

Reply:
=> https://github.com/DataDog/datadog-agent/pull/7342/commits/598217aa26b86eefbbe4703cc8ddbe52465aeb8f